pydantic/_internal/_model_construction.py

```diff
@@ -69,19 +77,28 @@ def inspect_namespace(namespace: dict[str, Any]) -> dict[str, ModelPrivateAttr]:
                 )
             private_attributes[var_name] = value
             del namespace[var_name]
-        elif not single_underscore(var_name):
+        elif var_name.startswith('__'):
             continue
         elif var_name.startswith('_'):
-            if var_name in raw_annotations and is_classvar(raw_annotations[var_name]):
-                continue
-            private_attributes[var_name] = PrivateAttr(default=value)
-            del namespace[var_name]
-        elif var_name not in raw_annotations and not isinstance(value, IGNORED_TYPES):
-            warnings.warn(
-                f'All fields must include a type annotation; '
-                f'{var_name!r} looks like a field but has no type annotation.',
-                DeprecationWarning,
-            )
+            if var_name in raw_annotations and not is_classvar(raw_annotations[var_name]):
+                private_attributes[var_name] = PrivateAttr(default=value)
+                del namespace[var_name]
+        elif var_name in base_class_vars:
+            continue
+        elif var_name not in raw_annotations and not isinstance(value, all_non_field_types):
+            if var_name in base_class_fields:
+                raise PydanticUserError(
+                    f'Field {var_name!r} defined on a base class was overridden by a non-annotated attribute. '
+                    f'All field definitions, including overrides, require a type annotation.',
+                )
+            elif isinstance(value, FieldInfo):
+                raise PydanticUserError(f'Field {var_name!r} requires a type annotation')
+            else:
+                raise PydanticUserError(
+                    f'A non-annotated attribute was detected: `{var_name} = {value!r}`. All model fields require a '
+                    f'type annotation; if {var_name!r} is not meant to be a field, you may be able to suppress this '
+                    f'warning by annotating it as a ClassVar or updating model_config["non_field_types"].',
```

Comment:
```suggestion
                    f'error by annotating it as a ClassVar or updating model_config["non_field_types"].',
```

Reply:
`PrivateAttr` wouldn't be allowed here specifically because you only get this error if the field name doesn't start with an `_`. 

However, while investigating this, I did discover that currently on this branch if you do:
```python
from pydantic import BaseModel, PrivateAttr

class WarningModel(BaseModel):
    _x = 1
```
You don't get an error _and_ `_x` doesn't become a `PrivateAttr`.

@samuelcolvin I'm assuming we want to make this an error with a similar message except that it also calls out `PrivateAttr` as an option. Does that make sense to you? Would you prefer we make note that `PrivateAttr` is another option if they change the name of the field?

Reply:
don't really care, if it's easy, match V1.